services/storage/virtualharddisk/virtualharddisk.go

```diff
@@ -62,7 +62,11 @@ func getWssdVirtualHardDisk(c *storage.VirtualHardDisk, groupName, containerName
 		} else {
 			wssdvhd.DiskFileFormat = common.DiskFileFormat_DiskFileFormatVHDX
 		}
-
+		if &c.CloudInitDataSource != nil {
```

Comment:
This condition will always be true, isn't it? You are comparing address of a variable against nil pointer. Why do we need this check?

Reply:
considering backwards compatibility since it's possible it might not be specified from moc-sdk-for-go when mocctl creates a storage.VirtualHardDisk directly